### DISCUSSION TITLE
Custom name for Foreign Key.

### DISCUSSION THREAD

**Discussion by nbowisdar** [piccolo-orm/piccolo_admin#358] [Q&A] [answered]
Hello, is it possible to set a name instead of fk numbers? 
![image](https://github.com/piccolo-orm/piccolo_admin/assets/78906490/aeabd695-c4b4-4d32-9f34-b298e6e943a1)
It would be pretty convenient if the name changed after updating __str__ method 
![image](https://github.com/piccolo-orm/piccolo_admin/assets/78906490/9a5092ef-568c-4c31-8941-1682bd16a484)

Thank you.

**Comment by dantownsend** [comment, marked as answer]
Yes, it's possible.

Piccolo doesn't use the `__str__` method to determine how the row is displayed. Instead it looks for a `get_readable` method:

https://piccolo-orm.readthedocs.io/en/latest/piccolo/schema/advanced.html#readable

Here's an example:

```python
# tables.py
from piccolo.table import Table
from piccolo.columns import Varchar
from piccolo.columns.readable import Readable


class Band(Table, tablename="music_band"):
    name = Varchar(length=100)

    @classmethod
    def get_readable(cls):
        return Readable(template="%s", columns=[cls.name])
```

I had a look, and we don't seem to mention it in the Piccolo Admin docs, only the main Piccolo docs, so this needs changing.

The reason we use `get_readable` is this is actually run in the database, so it's more efficient that calling `__str__` on each row.

**Reply by nbowisdar** [reply]
Thank you very much!

**Reply by Day0Dreamer** [reply]
Please provide suggestions for Foreign keys.

```
class ApplicationVersion(Table):
    id = Serial(primary_key=True)
    application = ForeignKey(references=Application, required=True)
    version_string = Varchar(length=255, index=True)
    release_date = Date()
    title = Varchar(length=255)

    @classmethod
    def get_readable(cls):
        return Readable(
            template="%s: %s", columns=[cls.application, cls.version_string]
        )
```

Given that in the Application table there is an entry with `id:1` and `name:application-name`,
Admin panel shows: `1: version-string` - as in an id for a foreign key
instead of `application-name: version-string`

The [documentation](https://piccolo-orm.readthedocs.io/en/latest/piccolo/api_reference/index.html#piccolo.table.Table.get_related) is not much of help here for me personally.

Please advise.

**Reply by Day0Dreamer** [reply]
Nevermind. This is the solution

```
class ApplicationVersion(Table):
    id = Serial(primary_key=True)
    application = ForeignKey(references=Application, required=True)
    version_string = Varchar(length=255, index=True)
    release_date = Date()
    title = Varchar(length=255)

    @classmethod
    def get_readable(cls):
        return Readable(
            template="%s: %s", columns=[cls.application.name, cls.version_string]
        )
```

Documentation could use some more examples I suppose
